src/libertem_live/detectors/merlin/data.py

```diff
@@ -663,13 +677,54 @@ def stream(self, num_frames=None, chunk_size=11):
         self.socket.read_headers()
         hdr = self.socket.get_acquisition_header()
         logger.info(hdr)
+
+        sig_shape = self.validate_get_sig_shape(hdr, self._sig_shape)
+
         pool = self.pool.get_impl(
             read_upto_frame=num_frames,
             chunk_size=chunk_size,
+            sig_shape=sig_shape,
         )
         with pool:
             while True:
                 with pool.get_result() as res_wrapped:
                     if pool.should_stop():
                         break
                     yield res_wrapped
+
+    def validate_get_sig_shape(self, hdr, sig_shape=None):
+        assembly_size = hdr.get('Assembly Size (NX1, 2X2)')
+        if assembly_size:
+            assembly_size = assembly_size.strip()
```

Comment:
Maybe like this, to also support the "gapped" assemblies?

```suggestion
            assembly_size = assembly_size.strip(" G")
```

Reply:
Thx! :-) Let's see when we get actual data of that kind to test.